### PR TITLE
replace react-native-animatable with react-native-reanimated

### LINKED ISSUE #2597: Find and replace implementation of react-native-animatable as it's an outdated package no longer been
Find and replace implementation of react-native-animatable as it's an outdated package no longer been

Note: this is research and testing only at this stage, one option is to just fork animatable repo and upgrade that to latest packages

### PR DESCRIPTION
**Pull request by aliseyalvi**
### What does this PR?
This PR replaces react-native-animatable with react-native-reanimated 2
This is WIP and initially replaced it inside toastNotification component

### Issue number
fixes #2597 

### Screenshots/Video

https://user-images.githubusercontent.com/48380998/213959508-eb4c2e83-518f-4ad8-bea7-294aa9c25832.mp4

### CODE REVIEW

**Comment by aliseyalvi** [comment]
@noumantahir replaced all implementations of react-native-animatable with react-native-reanimated except following 5 occurence because they would require some custom solution as we can't control the layout animation (entering/exiting) in react-native-reanimated 
image

File: src/components/organisms/quickProfileModal/children/profileStats.tsx [hunk 1]
```diff
@@ -39,7 +39,7 @@ export const ProfileStats = ({ data, horizontalMargin, intermediate }: Props) =>
 const StatItem = (props: { label: string; value: number | string; intermediate: boolean }) => (
   <View style={{ alignItems: 'center', flex: 1 }}>
     {!props.intermediate ? (
-      <Animated.Text animation="bounceIn" style={styles.statValue}>
+      <Animated.Text entering={BounceIn} style={styles.statValue}>
```
Comment (noumantahir) [inline src/components/organisms/quickProfileModal/children/profileStats.tsx:42]: having exiting animation here would be great as well perhaps

File: src/components/tabbedPosts/view/scrollTopPopup.tsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ const ScrollTopPopup = ({
   }
 
   return (
-    <AnimatedView style={styles.popupContainer} animation="zoomIn" duration={300}>
+    <Animated.View style={styles.popupContainer} entering={ZoomIn}>
```
Comment (noumantahir) [inline src/components/tabbedPosts/view/scrollTopPopup.tsx:29]: adding existing here as well

File: src/components/uploadsGalleryModal/children/uploadsGalleryContent.tsx [hunk 3]
```diff
@@ -273,9 +273,8 @@ const UploadsGalleryContent = ({
           isLoading={isDeleting}
         />
       ) : (
-        <AnimatedView
-          animation={deleteIds.length > 0 ? 'slideInRight' : 'slideOutRight'}
-          duration={300}
+        <AnimatedView.View
+          entering={deleteIds.length > 0 ? SlideInRight : SlideOutRight}
```
Comment (noumantahir) [inline src/components/uploadsGalleryModal/children/uploadsGalleryContent.tsx:277]: this needs updating, existing animation not working, set entering and exiting animations, and wrap the component around `deleteIds > 0 ` that should set things right

File: src/screens/editor/children/postOptionsModal.tsx [hunk 4]
```diff
@@ -155,13 +154,13 @@ const PostOptionsModal = forwardRef(
                 />
 
                 {scheduleLater && (
-                  <AnimatedView animation="flipInX" duration={700}>
+                  <Animated.View entering={FlipInEasyX.duration(700)}>
```
Comment (noumantahir) [inline src/screens/editor/children/postOptionsModal.tsx:157]: try existing here as well, might not working, but worth trying out

File: src/screens/transfer/screen/delegateScreen.js [hunk 5]
```diff
@@ -535,7 +535,7 @@ class DelegateScreen extends Component {
     );
 
     const _renderStepTwo = () => (
-      <AnimatedView animation="bounceInRight" delay={500} useNativeDriver>
+      <Animated.View entering={BounceInRight.duration(500)}>
```
Comment (noumantahir) [inline src/screens/transfer/screen/delegateScreen.js:538]: is adding delay() here required or was a typo before...?

**Review by noumantahir — CHANGES_REQUESTED** [review]
overall working great, just some small modifications

File: src/screens/transfer/screen/delegateScreen.js (hunk 5 above)
Comment (aliseyalvi) [reply src/screens/transfer/screen/delegateScreen.js:538]: yes, I think its a typo. Added delay again

**Review by noumantahir — APPROVED** [review]

**Review by feruzm — APPROVED** [review]
Lgtm

**Pull request merged**